### PR TITLE
Update notebook

### PR DESCRIPTION
**Pull request by mrchtr**

### CODE REVIEW

File: notebook.ipynb [hunk 1]
```diff
@@ -291,47 +231,18 @@
...
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "If you save the pipeline into a python file you can execute the pipeline using the Fondant cli as well.\n",
+    "compiler = DockerCompiler()\n",
+    "runner = DockerRunner()\n",
     "\n",
-    "> `fondant run local pipeline.py`"
+    "compiler.compile(pipeline=pipeline, output_path = \"docker-compose.yml\")\n",
+    "DockerRunner().run(\"docker-compose.yml\")"
```
Comment (RobbeSneyders) [inline notebook.ipynb:245]: We already specify `runner = DockerRunner()` above.

For me it's fine to keep this in two lines:
```python
DockerCompiler().compile(pipeline, output_path="docker-compose.yaml")
DockerRunner().run("docker-compose.yaml")
```

**Review by RobbeSneyders — COMMENTED** [review]
Can you still add the explorer to the notebook as well?

**Comment by PhilippeMoussalli** [comment]
> Can you still add the explorer to the notebook as well?

You can keep it similar to [this](https://github.com/ml6team/fondant-usecase-controlnet/blob/0e8322db133f2dcdbb1a395eadfe7ca0b740c47f/notebook.ipynb)

**Comment by mrchtr** [comment]
> > Can you still add the explorer to the notebook as well?
> 
> You can keep it similar to [this](https://github.com/ml6team/fondant-usecase-controlnet/blob/0e8322db133f2dcdbb1a395eadfe7ca0b740c47f/notebook.ipynb)

Thanks! I took over your implementation.

**Comment by RobbeSneyders** [comment]
Could you
- Resolve the merge conflicts
- Link the notebook in the README

File: notebook.ipynb [hunk 2]
```diff
@@ -257,29 +215,11 @@
...
-      " ⠋ Container weaviate-contextionary-1  Stopping                            \u001b[34m0.1s \u001b[0m\n",
-      " ⠋ Container weaviate-weaviate-1       Stopping                            \u001b[34m0.1s \u001b[0m\n",
-      "\u001b[?25h"
-     ]
-    }
-   ],
+   "outputs": [],
    "source": [
-    "!docker-compose -f weaviate/docker-compose.yaml up"
+    "!docker-compose -f weaviate/docker-compose.yaml -d up"
```
Comment (RobbeSneyders) [inline notebook.ipynb:222]:
```suggestion
    "!docker compose -f weaviate/docker-compose.yaml up --detach"
```
`docker-compose` is outdated

**Comment by PhilippeMoussalli** [comment]
> The pipeline fails on the base path:
> 
> ```
> ingestion-pipeline-chunk_text-1                 | FileNotFoundError: [Errno 2] No such file or directory: '/component/src/data-dir/ingestion-pipeline/ingestion-pipeline-20231106133647/load_from_huggingface_hub/manifest.json'
> ```
> 
> I think we need to set an absolute base path (we should be able to create this dynamically) and probably attach it as a volume when running the pipeline.

I handled this in my examples 
https://github.com/ml6team/fondant-usecase-controlnet/blob/0e8322db133f2dcdbb1a395eadfe7ca0b740c47f/notebook.ipynb

**Comment by mrchtr** [comment]
> The pipeline fails on the base path:
> 
> ```
> ingestion-pipeline-chunk_text-1                 | FileNotFoundError: [Errno 2] No such file or directory: '/component/src/data-dir/ingestion-pipeline/ingestion-pipeline-20231106133647/load_from_huggingface_hub/manifest.json'
> ```
> 
> I think we need to set an absolute base path (we should be able to create this dynamically) and probably attach it as a volume when running the pipeline.

The data-dir wasn't created during the pipeline execution. For now, I have added the code from @PhilippeMoussalli that creates the directory if it dosen't exists. But it looks like a general issue. Usually it should work even if the directory isn't there?

**Comment by RobbeSneyders** [comment]
> The data-dir wasn't created during the pipeline execution. For now, I have added the code from @PhilippeMoussalli that creates the directory if it dosen't exists. But it looks like a general issue. Usually it should work even if the directory isn't there?

The issue was that each component was handling the base path as relative to its root path. So it translated to `<component>/src/data_dir`, which is different for each component.

**Comment by RobbeSneyders** [comment]
Ok, everything seems to work now.
I would just add a cleanup section at the end running the following command:
```shell
!docker compose -f weaviate/docker-compose.yaml down
```

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+fondant[docker]==0.6.2
```
Comment (PhilippeMoussalli) [inline requirements.txt:1]: can we remove the `docker` extras, Its only needed for publishing image which is not needed here

File: src/notebook.ipynb [hunk 4]
```diff
@@ -35,63 +38,9 @@
...
-      "\n",
-      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m23.0.1\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m23.3.1\u001b[0m\n",
-      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49mpip install --upgrade pip\u001b[0m\n"
-     ]
-    }
-   ],
+   "outputs": [],
    "source": [
     "# Setup your environment \n",
     "!pip install \"fondant[docker]==0.6.2\""
```
Comment (RobbeSneyders) [inline src/notebook.ipynb:46]:
```suggestion
    "!pip install \"fondant==0.6.2\""
```

**Review by RobbeSneyders — APPROVED** [review]
Thanks @mrchtr!

**Pull request merged**